setup_builder.py

```diff
@@ -336,8 +337,9 @@ def _build_out_sections(
     packages = bsec.packages()
     if packages:
         cfg["options"]["packages"] = list_to_dangling(packages)
-    # else: this uses setuptools autodetection
-    #       https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#automatic-discovery
+    else:
+        cfg["options"]["packages"] = 'find:'
+        cfg["options.packages.find"]["exclude"] = list_to_dangling(DEFAULT_DIRECTORY_EXCLUDE)
```

Comment:
I think you need to create `cfg["options.packages.find"]` first

Reply:
Yes, I suppose so.  How about now?

Reply:
LGTM. You'd think python would've made this allowable